### PR TITLE
service/dap: add panic and throw text to stopped event

### PR DESCRIPTION
**Pull request by suzmue**
We can add more information to the stopped events on errors using
the `Text` field in the stopped event. We already use this to display
the runtime errors. Adding this information to the stopped reason will
also help to show the user additional info when a stopped event is not
associated with a particular goroutine.

### CODE REVIEW

**Comment by suzmue** [comment]
Example for deadlock:
 
Screen Shot 2021-06-30 at 3 31 33 PM

File: pkg/proc/eval.go [hunk 1]
```diff
@@ -67,16 +67,18 @@ func ConvertEvalScope(dbp *Target, gid, frame, deferCall int) (*EvalScope, error
...
-		return ThreadScope(ct)
-	}
 
 	var opts StacktraceOptions
 	if deferCall > 0 {
 		opts = StacktraceReadDefers
 	}
 
-	locs, err := g.Stacktrace(frame+1, opts)
+	var locs []Stackframe
```
Comment (aarzilli) [inline pkg/proc/eval.go:76]: I'm guessing that the reason for this change is that there is a bug where the frame parameter is ignored if we can't find the requested goroutine. I don't think we should sneak this bug fix in here without any comment.
It should be in a separate PR, if possible with a test in pkg/proc/proc_test.go.

Comment (suzmue) [reply pkg/proc/eval.go:76]: Done in #2563

**Review by aarzilli — APPROVED** [review]
LGTM

**Review by derekparker — CHANGES_REQUESTED** [review]
Blocking this from merge until https://github.com/go-delve/delve/pull/2563 is pulled in and this is rebased.

**Comment by suzmue** [comment]
> Blocking this from merge until #2563 is pulled in and this is rebased.

This branch has been rebased.

File: service/dap/server.go [hunk 2]
```diff
@@ -2509,34 +2509,19 @@ func (s *Server) onExceptionInfoRequest(request *dap.ExceptionInfoRequest) {
...
-			// Attempt to get the value of the throw reason.
-			// This is not currently working for Go 1.16 or 1.17: https://github.com/golang/go/issues/46425.
-			handleError := func(err error) {
-				if err != nil {
-					body.Description = fmt.Sprintf("Error getting throw reason: %s", err.Error())
-				}
+			body.Description, err = s.throwReason(goroutineID)
+			if err != nil {
+				body.Description = fmt.Sprintf("Error getting throw reason: %s", err.Error())
+				// This is not currently working for Go 1.16 or 1.17: https://github.com/golang/go/issues/46425.
```
Comment (polinasok) [inline service/dap/server.go:2515]: This issue was closed 7 days ago.

File: service/dap/server.go [hunk 3]
```diff
@@ -2593,6 +2578,25 @@ func (s *Server) onExceptionInfoRequest(request *dap.ExceptionInfoRequest) {
...
+func (s *Server) throwReason(goroutineID int) (string, error) {
+	return s.getExprString("s", goroutineID, 1)
+}
+
+func (s *Server) panicReason(goroutineID int) (string, error) {
+	return s.getExprString("(*msgs).arg.(data)", goroutineID, 0)
+}
+
+func (s *Server) getExprString(expr string, goroutineID, frame int) (string, error) {
+	exprVar, err := s.debugger.EvalVariableInScope(goroutineID, frame, 0, expr, DefaultLoadConfig)
```
Comment (polinasok) [inline service/dap/server.go:2590]: Do we need a larger string limit here?

Comment (suzmue) [reply service/dap/server.go:2590]: The throw reasons are all pretty short, the only concern is a long panic message. I think the default of 512 will probably be ok for now? We can definitely increase it if we need to.

File: service/dap/server.go (hunk 2 above)
Comment (suzmue) [reply service/dap/server.go]: Removed comment about 1.17

**Review by polinasok — APPROVED** [review]

**Review by derekparker — APPROVED** [review]
LGTM
